src/main/java/org/opentripplanner/model/plan/SearchWindowUtils.java

```diff
@@ -0,0 +1,102 @@
+package org.opentripplanner.model.plan;
+
+import java.time.Duration;
+import java.time.Instant;
+
+
+public class SearchWindowUtils {
+    /**
+     * Extra time is added to the search-window for the next request if the current
+     * result have few itineraries:
+     * <pre>
+     *  0: 6h
+     *  1: 4h
+     *  2: 2h
+     *  3: 1h
+     *  4: 30m
+     *  5: 20m
+     *  6: 10m
+     * </pre>
+     * Note!
+     *
+     * Unit: minutes
+     */
+    private static final int[] EXTRA_TIME_ITINERARIES = new int[]{
+            //0,   1,   2,  3,  4,  5,  6
+            360, 240, 120, 60, 30, 20, 10
+    };
+
+    /**
+     * We look at the search data after the trip search to see if we should adjust the search-
+     * window. This is done to avoid short-search-windows in low frequency areas, where the client
+     * would need to request the
```

Comment:
```suggestion
     * We look at the search data after the trip search to see if we should adjust the search
     * window. This is done to avoid short search windows in low frequency areas, where the client
     * would need to do multiple new request to fetch the next trips.
```